### PR TITLE
Differentiate between slow regression tests and performance experiments

### PR DESCRIPTION
**Pull request by wolfs**
Performance experiments should only run once a week, while slow performance regression should run at least every day if not more often.

### CODE REVIEW

File: .teamcity/Gradle_Check/model/CIBuildModel.kt [hunk 1]
```diff
@@ -396,15 +396,18 @@ enum class TestType(val unitTests: Boolean = true, val functionalTests: Boolean
...
     historical("FullPerformanceTest", 2280, "2.14.1,3.5.1,4.0,last", "--checks none");
 
-    fun asId(model: CIBuildModel): String {
-        return "${model.projectPrefix}Performance${name.capitalize()}Coordinator"
-    }
+    fun asId(model: CIBuildModel): String =
+        "${model.projectPrefix}Performance${name.capitalize()}Coordinator"
+
+    fun asUuid(model: CIBuildModel): String =
+        uuid?.let { model.projectPrefix + it } ?: asId(model)
```
Comment (blindpirate) [inline .teamcity/Gradle_Check/model/CIBuildModel.kt]: I would prefer `uuid:String = ""` and 

```
if(uuid.isEmpty()) asId(model) else "${model.projectPrefix}$uuid"
```

File: build.gradle.kts [hunk 2]
```diff
@@ -97,6 +97,10 @@ buildTypes {
         tasks("performance:distributedPerformanceTest")
     }
 
+    create("distributedSlowPerformanceTests") {
+        tasks("performance:distributedSlowPerformanceTest")
+    }
+
     create("distributedPerformanceExperiments") {
         tasks("performance:distributedPerformanceExperiment")
```
Comment (blindpirate) [inline build.gradle.kts]: This is not used any more I guess?

File: buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/performance/PerformanceTestPlugin.kt [hunk 3]
```diff
@@ -258,7 +273,15 @@ class PerformanceTestPlugin : Plugin<Project> {
...
+                excludeCategories(slowPerformanceRegressionTestCategory)
+            }
+            channel = "commits"
+            retryFailedScenarios()
+        }
+        create("distributedSlowPerformanceTest", DistributedPerformanceTest::class) {
+            (options as JUnitOptions).includeCategories(slowPerformanceRegressionTestCategory)
             channel = "commits"
             retryFailedScenarios()
         }
```
Comment (blindpirate) [inline buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/performance/PerformanceTestPlugin.kt]: And `create("distributedPerformanceExperiment")` can also be removed right?

File: .teamcity/Gradle_Check/model/CIBuildModel.kt (hunk 1 above)
Comment (wolfs) [reply .teamcity/Gradle_Check/model/CIBuildModel.kt]: I prefer using null, so I don't need to keep in mind that the empty string needs special handling.

File: build.gradle.kts (hunk 2 above)
Comment (wolfs) [reply build.gradle.kts]: I was going to leave the experiment tasks in, given that I thought about creating a configuration which can be run on demand with failure checks enabled.

Hmm, maybe we should remove the experiments from the historical performance tests, given that they don't use the baselines anyway, and move then to a separate job which has the checks enabled and runs once a week as well. WDYT?

File: buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/performance/PerformanceTestPlugin.kt (hunk 3 above)
Comment (wolfs) [reply buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/performance/PerformanceTestPlugin.kt]: See https://github.com/gradle/gradle/pull/10801#discussion_r326491920

File: build.gradle.kts (hunk 2 above)
Comment (blindpirate) [reply build.gradle.kts]: What do you mean by "which can be run on demand"? This configuration `distributedPerformanceExperiments` can't be invoked at all from TeamCity,

Comment (blindpirate) [reply build.gradle.kts]: I would keep them in historical performance tests, maybe we need to change the name to "full performance tests", same as the configuration name?

**Pull request merged**